BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt

```diff
@@ -160,7 +161,7 @@ open class BraintreeClient @VisibleForTesting internal constructor(
     open fun getConfiguration(callback: ConfigurationCallback) {
         getAuthorization { authorization, authError ->
             if (authorization != null) {
-                configurationLoader.loadConfiguration(authorization) { configuration, configError ->
+                configurationLoader.loadConfiguration(authorization, this) { configuration, configError ->
```

Comment:
**Random Thought**

We should consider delaying this configuration fetch. On Android, `sendAnalytics()` sends the event data to a SQLite store and schedules it to upload 30 seconds later. Fetching the configuration right before the scheduled upload would be an improvement on our current design.

Reply:
This is a good suggestion. We implemented it on iOS here - https://github.com/braintree/braintree_ios/pull/1337